source/adapters/level_zero/CMakeLists.txt

```diff
@@ -22,8 +22,25 @@ endif()
 if (NOT DEFINED LEVEL_ZERO_LIBRARY OR NOT DEFINED LEVEL_ZERO_INCLUDE_DIR)
     message(STATUS "Download Level Zero loader and headers from github.com")
 
+    # Workaround warnings/errors for Level Zero build
+    set(CMAKE_CXX_FLAGS_BAK "${CMAKE_CXX_FLAGS}")
+    if (WIN32)
+        # FIXME: Level Zero build fails with /DUNICODE
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /UUNICODE")
+        # USE_Z7 forces use of /Z7 instead of /Zi which is broken with sccache
+        set(USE_Z7 ON)
+    else()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-but-set-variable")
```

Comment:
Instead of listing all the warnings from L0, as they might change with the compiler or when L0 itself changes, we changed this to `$<$<IN_LIST:$<CXX_COMPILER_ID>,GNU;Clang;Intel;IntelLLVM>:-Wno-error>`. This way, the build doesn't fail on L0 warnings and we don't have to keep this list in sync. If we don't want to see the warnings, we can just also set `-w` to disable all warnings.

Reply:
thanks @pbalcer. Didnt understand. Do you mean that this line we have below:

```cpp
$<$<IN_LIST:$<CXX_COMPILER_ID>,GNU;Clang;Intel;IntelLLVM>:-Wno-error>
```

should already take care of this? Because that's not the case. If I dont add my changes, then intel/llvm compilation fails when using this UR commit.